packages/calcite-components/src/components/date-picker-month/date-picker-month.tsx

```diff
@@ -487,30 +574,143 @@ export class DatePickerMonth {
+      if (!!this.startDate && !!this.endDate) {
+        return isStart
+          ? date < this.endDate &&
+              (isDateAfterStartDateAndBeforeStart || isDateBeforeStartDateAndAfterStart)
+          : isDateBeforeEndDateAndAfterEnd || isDateAfterEndDateAndBeforeEnd;
+      } else if (!!this.startDate && !this.endDate) {
+        return isStart ? isDateBeforeStartDateAndAfterStart : isDateAfterStartDateAndBeforeEnd;
+      } else if (!this.startDate && !!this.endDate) {
+        return isStart ? isDateBeforeEndDateAndAfterStart : isDateAfterEndDateAndBeforeEnd;
+      }
+    } else {
+      if (!!this.startDate && !!this.endDate) {
+        return isStart ? isDateBeforeStartDateAndAfterStart : isDateAfterEndDateAndBeforeEnd;
+      }
+    }
+  }
+
+  private getDays = (
+    prevMonthDays: number[],
+    currMonthDays: number[],
+    nextMonthDays: number[],
+    position: "start" | "end" = "start",
+  ): Day[] => {
+    let month = this.activeDate.getMonth();
+    month = position === "end" ? month + 1 : month;
+    let dayInWeek = 0;
+    const getDayInWeek = () => dayInWeek++ % 7;
+    const year = this.activeDate.getFullYear();
+
+    const days: Day[] = [
+      ...prevMonthDays.map((day) => {
+        return {
+          active: false,
+          day,
+          dayInWeek: getDayInWeek(),
+          date: new Date(year, month - 1, day),
+        };
+      }),
+      ...currMonthDays.map((day) => {
+        const date = new Date(year, month, day);
+        const isCurrentDay = sameDate(date, new Date());
+        const active =
+          this.focusedDate &&
+          this.focusedDate !== this.startDate &&
+          this.focusedDate !== this.endDate
+            ? sameDate(date, this.focusedDate)
+            : sameDate(date, this.startDate) || sameDate(date, this.endDate);
+
+        return {
+          active,
+          currentMonth: true,
+          currentDay: isCurrentDay,
+          day,
+          dayInWeek: getDayInWeek(),
+          date,
+          ref: true,
+        };
+      }),
+      ...nextMonthDays.map((day) => {
+        return {
+          active: false,
+          day,
+          dayInWeek: getDayInWeek(),
+          date: new Date(year, month + 1, day),
+        };
+      }),
+    ];
+
+    return days;
+  };
+
+  private renderMonthCalendar(weekDays: string[], days: Day[], isEndCalendar = false): VNode {
+    return (
+      <div class="month" onKeyDown={this.keyDownHandler}>
+        <div class="week-header-container" role="row">
+          {weekDays.map((weekday) => (
+            <span class="week-header" role="columnheader">
+              {weekday}
+            </span>
+          ))}
+        </div>
+
+        <div class="week-days" role="row">
+          {days.map((day, index) => this.renderDateDay(day, isEndCalendar ? 50 + index : index))}
+        </div>
+      </div>
+    );
+  }
+
+  private monthHeaderSelectChange = (event: CustomEvent<Date>): void => {
+    const date = new Date(event.detail);
```

Comment:
Can you stop propagation of this internal event? Applies to other internal event handlers.

Reply:
This is unresolved. Ignore if bubbling is intentional (maybe add a comment to make this clear).

Reply:
Bubbling isn't required. Event propagation is stopped.